### PR TITLE
Sort the fields of record projection during normalization

### PR DESCRIPTION
**Pull request by sjakobi**
…as standardized in https://github.com/dhall-lang/dhall-lang/pull/625.

### CODE REVIEW

File: dhall/src/Dhall/Set.hs [hunk 1]
```diff
@@ -66,3 +67,6 @@ empty = Set Data.Set.empty Data.Sequence.empty
 difference :: Ord a => Set a -> Set a -> [a]
 difference os (Set s _) =
     filter (\ x -> not (Data.Set.member x s)) (toList os)
+
+sort :: Ord a => Set a -> Set a
```
Comment (Gabriel439) [inline dhall/src/Dhall/Set.hs]: Could you add a haddock comment since it's part of the public API?

Comment (sjakobi) [reply dhall/src/Dhall/Set.hs]: Done!

**Comment by sjakobi** [comment]
I also tried to fix `isNormalized`.

Did you ever take a stab at the property tests you mention in https://github.com/dhall-lang/dhall-haskell/pull/522#issuecomment-407105251, @Gabriel439?

**Comment by Gabriel439** [comment]
@sjakobi: I did, but they were disabled as part of the work on `Dhall.Eval`:

https://github.com/dhall-lang/dhall-haskell/blob/master/dhall/tests/Dhall/Test/QuickCheck.hs#L351-L354

Also: https://github.com/dhall-lang/dhall-haskell/pull/876#discussion_r275164324

**Comment by sjakobi** [comment]
@Gabriel439 Thanks for the pointers! I'll try to bring these tests back.

**Comment by Gabriel439** [comment]
@sjakobi: You're welcome! 🙂

**Pull request merged**